### DISCUSSION TITLE
upgrade next js 10   pragma and pragmaFrag cannot be set when runtime is automatic.

### DISCUSSION THREAD

**Discussion by Pirawat** [vercel/next.js#18440] [Help]
![Capture](https://user-images.githubusercontent.com/16607868/97526748-6270ef00-19dc-11eb-9b52-0a0c381f1161.JPG)

npm run build  error

**Comment by raymclee** [comment]
same issue with nextjs 10 and theme-ui

**Reply by lachlanjc** [reply]
Hey! We've added automatic runtime support (https://github.com/system-ui/theme-ui/pull/1237) to the latest (alpha) release of Theme UI. Try installing `theme-ui@next` & let us know in the Theme UI repo how it goes :)

**Reply by professeurb** [reply]
I have tried it, having the same kind of problem. Now, there is a problem with Prism:
error - ./node_modules/@theme-ui/prism/dist/prism.esm.js
Attempted import error: 'Styled' is not exported from 'theme-ui'.

**Reply by lachlanjc** [reply]
Ah yeah, we've renamed `Styled` to `Themed`. We'll release a stable soon with an upgrading guide & all.

**Reply by BenAOlson** [reply]
@lachlanjc So, with the alpha version, I should still be able to just do this?
```javascript
/** @jsx jsx */
import { jsx } from 'theme-ui
```

**Reply by lachlanjc** [reply]
@BenAOlson it depends if you're using the automatic runtime or not. If you are, you're looking for `jsxImportSource` instead: https://dev.theme-ui.com/sx-prop — otherwise you can keep using the same `jsx` syntax you mention. You can also skip all of it & use Theme UI Components (like Box & Text).

We unfortunately don't have very clear docs on this yet, but feel free to join the Discord (https://discord.gg/stp6BY7MXA) or ask in the Theme UI repo issues.

**Comment by Callenowy** [comment]
It's not a Nextjs nor Theme-UI issue. 
React 17 introduced a new version of JSX which has two runtime options "classic" and "automatic".
The old JSX transform turned JSX into `React.createElement(...)` calls, which has some disadvantages like for example require React to be in the scope when using JSX. 
While the new one (automatic) automatically imports special functions from those new entry points in the React package and calls them, so import React to use JSX is not needed anymore. 

Next.js v9.5.3+ uses the new transform for compatible React versions.

Read more here: https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html

**Reply by ishu202** [reply]
how did you solved it ?

**Reply by jdsghuman** [reply]
Add this to your files as a fix for now:

/** @jsxRuntime classic */
/** @jsx jsx */
import { jsx } from '@emotion/core'

**Reply by Callenowy** [reply]
> how did you solved it ?

I solved it by creating custom `.babelrc` file in project directory.

```javascript
{
  "presets": [
    [
      "next/babel",
      {
        "preset-react": {
          "runtime": "automatic",
          "importSource": "@emotion/react"
        }
      }
    ]
  ],
  "plugins": ["@emotion/babel-plugin"]
}
```

https://nextjs.org/docs/advanced-features/customizing-babel-config

**Reply by sujumayas** [reply]
Is there a similar babel preset / plugin for theme-ui ?

**Reply by sbin0819** [reply]
@jdsghuman 
Thanks!!!
One thing miss spell
/** @jsxRuntime classic / -> /** @jsxRuntime classic */

**Reply by ishu202** [reply]
anyone who cares to understand what this block is doing.
/** @jsxRuntime classic /
/* @jsx jsx */
import { jsx } from '@emotion/core'

1st line
setting the jsx runtime to classic meaning not to use default jsx import with next.
2nd line 
use the jsx which is being imported to from the import statement.
3rd line 
the import of the source.

i looked up the source code how it works internally. and it is very beautiful.

**Comment by ishu202** [comment]
there are two ways to solve the problem one is to define where jsx import is coming from i.e
add this line on the top.
/** @jsximport 'theme-ui' */   (if you are using the theme-ui. same goes for the emotion.)
or you could define the bable config and define you imports there.

/** Important */
the way jsx is transformed into the dom element has been changed so its worth checking out the the two modes react 17.x update introduced.
https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html
here is the link hope it helps. and answers anyone's question in future.
